src/test/java/javaslang/collection/AbstractTraversableOnceTest.java

```diff
@@ -0,0 +1,147 @@
+/*     / \____  _    ______   _____ / \____   ____  _____
+ *    /  \__  \/ \  / \__  \ /  __//  \__  \ /    \/ __  \   Javaslang
+ *  _/  // _\  \  \/  / _\  \\_  \/  // _\  \  /\  \__/  /   Copyright 2014-2015 Daniel Dietrich
+ * /___/ \_____/\____/\_____/____/\___\_____/_/  \_/____/    Licensed under the Apache License, Version 2.0
+ */
+package javaslang.collection;
+
+import org.assertj.core.api.*;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.stream.Collector;
+
+/**
+ * Tests all methods defined in {@link Traversable}.
+ */
+public abstract class AbstractTraversableOnceTest {
```

Comment:
This is a great idea to extract it! We have some many classes beside collections, which implement TraversableOnce already!